components/brave_ads/browser/ads_service.h

```diff
@@ -108,63 +108,14 @@ class AdsService : public KeyedService {
   // |base::Value::List| containing info of the obtained diagnostics.
   virtual void GetDiagnostics(GetDiagnosticsCallback callback) = 0;
 
-  // Called when the user changes the locale of their operating system. This
-  // call is not required if the operating system restarts the browser when
-  // changing the locale. |locale| should be specified in either
-  // <ISO-639-1>-<ISO-3166-1> or <ISO-639-1>_<ISO-3166-1> format.
-  virtual void OnLocaleDidChange(const std::string& locale) = 0;
-
   // Called when a resource component has been updated.
   virtual void OnDidUpdateResourceComponent(const std::string& id) = 0;
 
-  // Called when the page for |tab_id| has loaded and the content is available
-  // for analysis. |redirect_chain| containing a list of redirect URLs that
-  // occurred on the way to the current page. The current page is the last one
-  // in the list (so even when there's no redirect, there should be one entry in
-  // the list). |html| containing the page content as HTML.
-  virtual void OnTabHtmlContentDidChange(
-      const SessionID& tab_id,
-      const std::vector<GURL>& redirect_chain,
-      const std::string& html) = 0;
-
-  // Called when the page for |tab_id| has loaded and the content is available
-  // for analysis. |redirect_chain| containing a list of redirect URLs that
-  // occurred on the way to the current page. The current page is the last one
-  // in the list (so even when there's no redirect, there should be one entry in
-  // the list). |text| containing the page content as text.
-  virtual void OnTabTextContentDidChange(
-      const SessionID& tab_id,
-      const std::vector<GURL>& redirect_chain,
-      const std::string& text) = 0;
-
   // Called when a page navigation was initiated by a user gesture.
   // |page_transition_type| containing the page transition type, see enums for
   // |PageTransitionType|.
   virtual void TriggerUserGestureEvent(int32_t page_transition_type) = 0;
```

Comment:
Should we move `TriggerUserGestureEvent` to the observer?

Reply:
Added `NotifyUserGestureEventTriggered` notification